flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -498,13 +501,6 @@ public boolean isShutdown() {
             @Nullable String externalSavepointLocation,
             boolean isPeriodic) {
 
-        if (props.getCheckpointType().getPostCheckpointAction() == PostCheckpointAction.TERMINATE
-                && !(props.isSynchronous() && props.isSavepoint())) {
-            return FutureUtils.completedExceptionally(
-                    new IllegalArgumentException(
-                            "Only synchronous savepoints are allowed to advance the watermark to MAX."));
-        }
```

Comment:
I don't really get why it was removed?

Reply:
I find those checks at this locaiton pointless. Valid combinations of properties of a `SnapshotType` are limited by factory methods of `CheckpointType` & `SavepointType`. I see no reason for checking it again here.